libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt

```diff
@@ -153,8 +149,37 @@ constructor(
         "com.mapbox.navigation.trip.notification.MapboxTripNotification"
     private val MAPBOX_NOTIFICATION_ACTION_CHANNEL = "notificationActionButtonChannel"
 
-    init {
+    /**
+     * @param context applicationContext
+     * @param accessToken [Mapbox Access Token](https://docs.mapbox.com/help/glossary/access-token/)
+     */
+    constructor(context: Context, accessToken: String) : this(context, defaultNavigationOptions(context, accessToken))
+
+    /**
+     * @param context applicationContext
+     * @param accessToken [Mapbox Access Token](https://docs.mapbox.com/help/glossary/access-token/)
+     * @param locationEngine used to listen for raw location updates
+     */
+    constructor(
+        context: Context,
+        accessToken: String,
+        locationEngine: LocationEngine
+    ) : this(context, defaultNavigationOptions(context, accessToken), locationEngine)
 
+    /**
+     * @param context applicationContext
+     * @param accessToken [Mapbox Access Token](https://docs.mapbox.com/help/glossary/access-token/)
+     * @param locationEngine used to listen for raw location updates
+     * @param locationEngineRequest used to request raw location updates
+     */
+    constructor(
+        context: Context,
+        accessToken: String,
+        locationEngine: LocationEngine,
+        locationEngineRequest: LocationEngineRequest
+    ) : this(context, defaultNavigationOptions(context, accessToken), locationEngine, locationEngineRequest)
```

Comment:
I'd say that we don't need these overloads. It can be a slippery slope trying to support them when new parameters come into play.

I'd much rather expose only the primary constructor that has the `NavigationOptions` required, but also property document the `defaultNavigationOptions` method in the KDoc of that primary constructor so that developers have a better chance of discovering it.

Reply:
like it 👍